lib/xilinx/configuration_packet.cc

```diff
@@ -0,0 +1,220 @@
+#include <prjxray/xilinx/configuration_packet.h>
+
+#include <iomanip>
+#include <iostream>
+#include <ostream>
+
+#include <prjxray/bit_ops.h>
+
+namespace prjxray {
+namespace xilinx {
+
+template <>
+std::pair<absl::Span<uint32_t>,
+          absl::optional<ConfigurationPacket<Spartan6ConfigurationRegister>>>
+ConfigurationPacket<Spartan6ConfigurationRegister>::InitWithWords(
+    absl::Span<uint32_t> words,
+    const ConfigurationPacket<Spartan6ConfigurationRegister>* previous_packet) {
+	using ConfigurationRegister = Spartan6ConfigurationRegister;
+	// Need at least one 32-bit word to have a valid packet header.
+	if (words.size() < 1)
+		return {words, {}};
+
+	uint32_t header_type = bit_field_get(words[0], 15, 13);
+	switch (header_type) {
+		case 0x0:
```

Comment:
Constants for 0x0/0x1/0x2 here?

Reply:
This code should be part of adding the Spartan6 support. But will make sure I use the enums there.